v2_bridges/erc20_on_int/src/int/metadata.rs

```diff
@@ -10,20 +10,29 @@ impl Erc20OnIntEthTxInfo {
     fn to_metadata(&self) -> Result<Metadata> {
         let user_data = if self.user_data.len() > MAX_BYTES_FOR_ETH_USER_DATA {
             info!(
-                "✘ `user_data` redacted from `Metadata` ∵ it's > {} bytes",
+                "`user_data` redacted from `Metadata` ∵ it's > {} bytes",
                 MAX_BYTES_FOR_ETH_USER_DATA
             );
             vec![]
         } else {
             self.user_data.clone()
         };
-        Ok(Metadata::new(
-            &user_data,
-            &MetadataAddress::new_from_eth_address(
-                &self.token_sender,
-                &MetadataChainId::from_str(&self.origin_chain_id.to_string())?,
-            )?,
-        ))
+
+        let mut origin_address = MetadataAddress::new_from_eth_address(
+            &self.token_sender,
+            &MetadataChainId::from_str(&self.origin_chain_id.to_string())?,
+        )?;
+
+        if cfg!(feature = "include-origin-tx-details") && self.metadata_event.is_some() {
+            let metadata = self
+                .metadata_event
+                .as_ref()
+                .expect("this not to fail due to preceeding line");
+
+            origin_address = MetadataAddress::new(metadata.origin_address(), metadata.origin_chain_id())?;
+        };
```

Comment:
I can't remember why I did it this mutable way originally but I now prefer to remove the `mut` since it's pointless, and just have that version in an `else`. I've also added a debug message to mine in the first `if` clause:

```rust
debug!("using origin tx details in metadata");
```

Have updated my PR too with the same change ([see here](https://github.com/pnetwork-association/ptokens-core/pull/18/commits/5b255e771a7d8f6a2293bb5b552c4bbd3fb03653)).